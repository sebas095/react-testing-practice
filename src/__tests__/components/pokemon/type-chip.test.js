import { mount } from "enzyme";
import TypeChip from "../../../components/pokemon/type-chip";

import "jest-styled-components";

describe("Type Chip", () => {
  it("mount ok", () => {
    const wrapper = mount(<TypeChip name="grass" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Shown in purple for poison type", () => {
    const wrapper = mount(<TypeChip name="poison" />);
    expect(wrapper).toHaveStyleRule("background-color", "purple");
  });

  it("Shown in green for grass type", () => {
    const wrapper = mount(<TypeChip name="grass" />);
    expect(wrapper).toHaveStyleRule("background-color", "green");
  });

  it("Shown in yellow for electric type", () => {
    const wrapper = mount(<TypeChip name="electric" />);
    expect(wrapper).toHaveStyleRule("background-color", "yellow");
  });
});
