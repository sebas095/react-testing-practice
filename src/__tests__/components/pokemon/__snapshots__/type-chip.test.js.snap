// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Chip mount ok 1`] = `ReactWrapper {}`;
